scripthost/src/main/kotlin/com/commonwealthrobotics/bowlerkernel/scripthost/Session.kt

```diff
@@ -0,0 +1,271 @@
+                            }
+
+                            send(it.input.build() as SessionServerMessage)
+
+                            // Also need to set the response handler so the callback will be called
+                            responseHandlers[id] = it.callback
+                        } else {
+                            send(it.input.build() as SessionServerMessage)
+                        }
+
+                        true
+                    }
+
+                    clientChannel.onReceiveOrClosed {
+                        when {
+                            it.isClosed -> false
+
+                            else -> it.value.let { msg ->
+                                logger.debug { "Received: $msg" }
+
+                                val requestId = when {
+                                    msg.hasConfirmationResponse() -> msg.confirmationResponse.requestId
+                                    msg.hasCredentialsResponse() -> msg.credentialsResponse.requestId
+                                    msg.hasTwoFactorResponse() -> msg.twoFactorResponse.requestId
+                                    msg.hasError() -> msg.error.requestId
+                                    else -> null
+                                }
+
+                                val result = when {
+                                    requestId != null -> {
+                                        responseHandlers.remove(requestId)?.invoke(msg)
+                                        Either.Right(Unit)
+                                    }
+
+                                    msg.hasRunRequest() -> runRequest(msg.runRequest, sessionChannel)
+                                    msg.hasCancelRequest() -> TODO("Not yet implemented")
+                                    msg.hasNewConfig() -> TODO("Not yet implemented")
+                                    else -> throw IllegalStateException("Unhandled message type: $msg")
+                                }
+
+                                logger.debug { "Result of handling client message: $result" }
+
+                                if (msg.hasRunRequest()) {
+                                    if (result is Either.Left) {
+                                        logger.warn(result.a) { "Script did not run successfully." }
+                                        send(
+                                            sessionServerMessage {
+                                                errorBuilder.requestId = msg.runRequest.requestId
+                                                errorBuilder.description =
+                                                    "Script did not run successfully: ${result.a.localizedMessage}"
+                                            }
+                                        )
+                                    }
+                                }
```

Comment:
Why do we only send an error on run requests? Shouldn't this be for all requests? (And shouldn't it be the same format for all requests?)

IMO it would be better to implement this (both the logging and the error response) within `runRequest`.